lib/logutils/logutils.go

```diff
@@ -0,0 +1,430 @@
+// Copyright (c) 2016-2017 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logutils
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log/syslog"
+	"os"
+	"path"
+	"runtime"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/mipearson/rfw"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// FilterLevels returns all the logrus.Level values <= maxLevel.
+func FilterLevels(maxLevel log.Level) []log.Level {
+	levels := []log.Level{}
+	for _, l := range log.AllLevels {
+		if l <= maxLevel {
+			levels = append(levels, l)
+		}
+	}
+	return levels
+}
+
+// Formatter is our custom log formatter, which mimics the style used by the Python version of
```

Comment:
Makes sense.  I'll make those changes.  Thanks for writing up the descriptive comment for me